### ISSUE TITLE
FAA0001 false positive on expressions containing Count()

### ISSUE THREAD

**Issue by davidnmbond** [fluentassertions/fluentassertions.analyzers#383]
### Description

FAA0001

### Complete minimal example reproducing the issue

```C#
		// Arrange
		int[] x = [1, 2, 3];
		int[] y = [4, 5, 6];
		var both = x.Concat(y).ToArray();

		// Assert
		(x.Length + y.Length).Should().Be(both.Length);
```

### Expected behavior:

No suggestion

### Actual behavior:

FAA0001 Use .Should().HaveCount()

### Versions

* Fluent Assertions Analyzers 0.33.0
* .NET runtime and version .NET 8.0

### Additional Information

None
